airbyte-integrations/connectors/source-bigquery/source_bigquery/source.py

```diff
@@ -117,5 +116,16 @@ def streams(self, config: Mapping[str, Any]) -> List[Stream]:
 
         :param config: A Mapping of the user input configuration as defined in the connector spec.
         """
-        pass
+        self._auth = BigqueryAuth(config)
+
+        if not self.streams_catalog:
```

Comment:
It can be a big slow to run the whole `discover` especially if the user only wants an handful of streams here. I don't think this is a priority to fix this but I still wanted to raise the flag. We have sources where we do [parallel processing](https://github.com/airbytehq/airbyte/blob/a8ec6d0f37bb45d290ae9beb32db6c8d030d0c85/airbyte-integrations/connectors/source-salesforce/source_salesforce/api.py#L371-L385) for that. I'm wondering if we have other solutions internally for that and I'll sync with our team.

Reply:
Fixed, I removed the override of the `discover`, now I just add my streams in `streams` method

Reply:
I think we agreed during a demo that even if a user defined `streams_catalog`, we need to return the other streams. Basically:
* streams mapping to discovered table should always be there
* if a `streams_catalog` is defined it should be return